### DISCUSSION TITLE
I'm working on a college project to train a basic reinforcement model, and decided to use UXsim for it

### DISCUSSION THREAD

**Discussion by Yateen00** [toruseo/UXsim#154] [Q&A] [answered]
My project requires me to be able to dynamically change which lane to turn green(so that the vehicles from that lane can move to any other) and how long to turn the signal green for at each step.

I've been struggling to achieve this. I've tried hard to understand the docs, especially the dqn one, which is similar to my project, but I'm struggling to understand what: 
signal group, 
signal paramter in add node and
 the bit string to set signal_phase and signal_t is. 
If someone could help me with how to achieve this or link to some resource which I can learn from, i would really appreciate it. Thank you for reading.

**Comment by toruseo** [comment, marked as answer]
Thanks for your interest!

Lets consider the following situation. A node is connected to 4 links, say link A, B, C, and D. The value of `signal_group` parameter of links A and C is 0, and that of B and D is 1. 
Then, links A and C will have green light when the node's `signal_phase` is 0, and B and D will be green when `signal_phase` is 1. The node's `signal` parameter will be something like `[60, 50]`, meaning that phase 0 is 60 seconds and phase 1 is 50 seconds.

> the bit string to set signal_phase and signal_t 

This is to convert one integer variable (output of DQN) to four binary variables (input for four signal nodes).

I recommend you to see and modify the following code to see what will happen if the parameter values were changed.
https://github.com/toruseo/UXsim/blob/main/demos_and_examples/example_10en_signal_4legged_intersection.py
